java/server/src/org/openqa/selenium/grid/graphql/Node.java

```diff
@@ -28,12 +31,18 @@
   private final URI uri;
   private final boolean isUp;
   private final int maxSession;
+  private final String capabilities;
 
-  public Node(UUID id, URI uri, boolean isUp , int maxSession) {
+  public Node(UUID id,
+              URI uri,
+              boolean isUp,
+              int maxSession,
+              Map<Capabilities, Integer> capabilities) {
     this.id = Require.nonNull("Node id", id);
     this.uri = Require.nonNull("Node uri", uri);
     this.isUp = isUp;
     this.maxSession = Require.nonNull("Node maxSession", maxSession);
+    this.capabilities = Require.nonNull("Node capabilities", new Json().toJson(capabilities));
```

Comment:
Nit: Make this a private constant in the class, as some of the state picked up by reflection is stored.